src/interfaces/address-interfaces/address-interface.ts

```diff
@@ -0,0 +1,22 @@
+// Base address Interface to match the address.json schema and it's optional fields.
+export interface AddressIF {
+  addressCity: string;
+  addressCountry: string;
+  addressRegion: string;
+  deliveryInstructions?: string;
+  postalCode: string;
+  streetAddress: string;
+  streetAddressAdditional?: string;
+}
+
+// Interface to define the joint base address response
+export interface BaseAddressObjIF {
+  deliveryAddress: AddressIF;
+  mailingAddress: AddressIF;
+}
+
+// Interface to define the Bcorps address response
+export interface BcorpAddressIf {
+  registeredOffice: BaseAddressObjIF;
+  recordsOffice: BaseAddressObjIF;
```

Comment:
No action item for you . But I will be changing the type name to  something like IncorporationAddressIF (because it is not really BcorpAddressIf alone) and make recordsOffice as an optional property.

Reply:
Totally, makes perfect sense to me! Now that i look at it, i didn't even implement this Interface and i just described the registered or records office as a baseAddressIF.